qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -991,13 +994,13 @@ def __init__(self, *, win_id, mode_manager, private, parent=None):
         self.caret = WebEngineCaret(mode_manager=mode_manager,
                                     tab=self, parent=self)
         self.zoom = WebEngineZoom(tab=self, parent=self)
-        self.search = WebEngineSearch(parent=self)
+        self.search = WebEngineSearch(tab=self, parent=self)
         self.printing = WebEnginePrinting(tab=self)
         self.elements = WebEngineElements(tab=self)
         self.action = WebEngineAction(tab=self)
         self.audio = WebEngineAudio(parent=self)
         self._permissions = _WebEnginePermissions(tab=self, parent=self)
-        self._scripts = _WebEngineScripts(tab=self, parent=self)
+        self.scripts = _WebEngineScripts(tab=self, parent=self)
```

Comment:
I'm not sure if we should make `scripts` public or just add a public function which updates the stylesheet.

We might also be able to use qt signals/slots for this, but it might make things more complicated.

Reply:
I originally [proposed](https://github.com/qutebrowser/qutebrowser/pull/4186#pullrequestreview-151872622) that, but you're right, that's not a good idea - it shouldn't e.g. be exposed via a future plugin API.

I rewrote things to use signals: 54c39484d3c981783055ebd69b66835cd1ee2430